### PR TITLE
Added possibility to show CodeceptJS config and added Copy to Clipboard

### LINKED ISSUE #4: Add generated selenium config for CodeceptJS/JSQA - frontend part
Connected with - https://github.com/dixons/mobile-manager/issues/31

### PR DESCRIPTION
**Pull request by CDJojn**
Web client part of this enhancement https://github.com/dixons/mobile-manager/issues/31

### CODE REVIEW

File: src/applications/devices/components/DeviceDetailBasicInfo.tsx [hunk 1]
```diff
@@ -6,11 +6,12 @@ import {DeviceDetailSelenium} from "./DeviceDetailSelenium";
 import { DeviceDetailActionButtons } from "./DeviceDetailActionButtons";
 import { IDevice } from "../../../_shared/api/models/IDevice";
 import { IReservationApplied } from "../../../_shared/api/models/IReservationApplied";
+import { ISeleniumConfig } from 'src/_shared/api/models/ISeleniumConfig';
 
 interface IDeviceDetailBasicInfoProps {
-  device: IDevice;
```
Comment (michalpaukert) [inline src/applications/devices/components/DeviceDetailBasicInfo.tsx:11]: why did you remove semicolon?

File: src/applications/devices/components/DeviceDetailSelenium.tsx [hunk 2]
```diff
@@ -1,32 +1,63 @@
...
+import * as copy from 'copy-to-clipboard';
 
 interface IDeviceDetailSeleniumProps {
-  seleniumConfig: string;
+  seleniumConfig: ISeleniumConfig;
 }
 
-export class DeviceDetailSelenium extends React.PureComponent<IDeviceDetailSeleniumProps> {
+interface IDeviceDetailSeleniumState {
+  shownSeleniumConfig: string;
```
Comment (michalpaukert) [inline src/applications/devices/components/DeviceDetailSelenium.tsx]: I prefer something like selectedSeleniumConfig

File: src/applications/devices/components/DeviceDetailSelenium.tsx [hunk 3]
```diff
@@ -1,32 +1,63 @@
...
+    this.state = {
+      shownSeleniumConfig: "xtest"
+    };
+
+    this.showConfig = this.showConfig.bind(this);
+    this.copyToClipboard = this.copyToClipboard.bind(this);
+  }
 
+  public render(): JSX.Element {
+    let data = "Loading...";
```
Comment (michalpaukert) [inline src/applications/devices/components/DeviceDetailSelenium.tsx]: Create loading component and return <Loading /> instead of text
Or you can simplify this by
```suggestion
const data = this.props.seleniumConfig && this.props.seleniumConfig[this.state.shownSeleniumConfig] || 'Loading...';
```

File: src/applications/devices/components/DeviceDetailSelenium.tsx [hunk 4]
```diff
@@ -1,32 +1,63 @@
...
-              className="device-detail__selenium-config"
-            >
-              {this.props.seleniumConfig}
-            </div>
+          <div>
+            <button className="device-detail__selenium-button" data-config="xtest" onClick={e => this.showConfig(e)}>PHP-xTest</button>
+            <button className="device-detail__selenium-button" data-config="codeceptjs" onClick={e => this.showConfig(e)}>CodeceptJS/JSQA</button>
+          </div>
+            <pre className="device-detail__selenium-config">
+              {data}}
```
Comment (michalpaukert) [inline src/applications/devices/components/DeviceDetailSelenium.tsx]:
```suggestion
              {data}
```

File: src/applications/devices/components/DeviceDetailSelenium.tsx [hunk 5]
```diff
@@ -1,32 +1,63 @@
...
-              <button disabled={true}>Download</button>
+              <button className="device-detail__selenium-button" onClick={e => this.copyToClipboard(e)}>Copy To Clipboard</button>
+              <button className="device-detail__selenium-button" disabled={true}>Download</button>
             </div>
           </div>
         </div>
       </div>
     )
   }
+  private showConfig(event: React.SyntheticEvent<HTMLButtonElement>): void {
```
Comment (michalpaukert) [inline src/applications/devices/components/DeviceDetailSelenium.tsx]: I prefer to use toggleConfig

**Review by michalpaukert — CHANGES_REQUESTED** [review]

File: src/applications/devices/components/DeviceDetailBasicInfo.tsx (hunk 1 above)
Comment (CDJojn) [reply src/applications/devices/components/DeviceDetailBasicInfo.tsx:11]: Fixed

File: src/applications/devices/components/DeviceDetailSelenium.tsx (hunk 2 above)
Comment (CDJojn) [reply src/applications/devices/components/DeviceDetailSelenium.tsx]: Fixed

File: src/applications/devices/components/DeviceDetailSelenium.tsx (hunk 3 above)
Comment (CDJojn) [reply src/applications/devices/components/DeviceDetailSelenium.tsx]: Fixed, simplified version used

File: src/applications/devices/components/DeviceDetailSelenium.tsx (hunk 4 above)
Comment (CDJojn) [reply src/applications/devices/components/DeviceDetailSelenium.tsx]: typo, fixed

File: src/applications/devices/components/DeviceDetailSelenium.tsx (hunk 5 above)
Comment (CDJojn) [reply src/applications/devices/components/DeviceDetailSelenium.tsx]: Fixed

**Review by michalpaukert — APPROVED** [review]

**Review by Stanin84 — APPROVED** [review]
